### ISSUE TITLE
Add a way to create GitHub Releases

### ISSUE THREAD

**Issue by DAmNRelentless** [Kir-Antipov/mc-publish#7]
Is there a way to publish the file as a GitHub pre-release?
I want to have it as a pre-release if it's an alpha or beta.

**Comment by Kir-Antipov** [comment]
This question is not related to the project, it's rather about GitHub Actions themselves, but ok.

It's up to you how to make this work:

 1) Make two different actions for releases and prereleases like so:
    ```yaml
    name: Prerelease
    on:
      release:
        types: [published]

    jobs:
      build:
          if: "github.event.release.prerelease"
          runs-on: ubuntu-latest
          steps:
            # ...
            - uses: Kir-Antipov/mc-publish@v2.1
              with:
                # ...
                version-type: alpha
    ```

    ```yaml
    name: Release
    on:
      release:
        types: [published]

    jobs:
      build:
          if: "!github.event.release.prerelease"
          runs-on: ubuntu-latest
          steps:
            # ...
            - uses: Kir-Antipov/mc-publish@v2.1
              with:
                # ...
                version-type: release
    ```

 2) As an alternative, you can provide a dynamically computed version to the action's input:
    ```yaml
    on:
      release:
        types: [published]

    jobs:
      build:
        # ...
        steps:
          # ...
          - uses: Kir-Antipov/mc-publish@v2.1
            with:
              # ...
              version-type: ${{ some_value_computed_earlier }}
    ```

**Comment by DAmNRelentless** [comment]
It is related to the project since I want to use mc-publish for releasing on CurseForge, Modrinth **and** GitHub.
I admit my question was not phrased very well. I wanted to know if your action publishes the file as a GitHub pre-release when the version is set to alpha or beta.

Right now, I am using an artifact release action for GitHub releases so I can define if it should be a pre-release or not.
Basically, if my mod version is an alpha or beta, it will be a GitHub pre-release, otherwise, it will be a normal release.

https://github.com/AlmostReliable/lazierae2-forge/blob/46eb4536ccd6fe44c384fb6f8720c273a3bc6613/.github/workflows/release.yml#L65-L124

**Comment by Kir-Antipov** [comment]
This action does not create a release (at least at the moment), because I originally meant it to be used in `release.types: [published]` environment, therefore changlelog and other things could be inferred from there; it just uploads given artifacts to the release that triggered the action.

**Comment by Kir-Antipov** [comment]
Actually, I can see your approach as a good use case. I think, I'll implement GitHub Release creation in upcoming v3.0

**Comment by Kir-Antipov** [comment]
And now it makes sense to answer your question - if `version-type` of your mod is not `release`, than created GitHub Release will be marked as `pre-release` automatically :) However, you can change this behavior via providing a `github-prerelease` input all by yourself

**Comment by DAmNRelentless** [comment]
Thanks a lot! Sorry again for the confusion but you got a new feature out of that, I mean that's something! :D

**Comment by Kir-Antipov** [comment]
Don't worry, there's nothing to be sorry about xD I don't even know why I didn't think about this feature in the first place ¯\\_(ツ)_/¯
